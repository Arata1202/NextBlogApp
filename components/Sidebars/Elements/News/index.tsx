'use client';
import { useTheme } from 'next-themes';
import { NewspaperIcon } from '@heroicons/react/24/solid';
import styles from './index.module.css';
import { news } from '@/section/news';

export default function News() {
  const { theme } = useTheme();
  return (
    <div className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}>
      <h1 className={`${styles.profile} text-2xl text-center font-semibold flex justify-center`}>
        <NewspaperIcon className="h-8 w-8 mr-2" aria-hidden="true" />
        お知らせ
      </h1>

      <div className="">
        <div className="mt-5" style={{ maxHeight: '240px', overflowY: 'auto' }}>
          <table className="min-w-full">
            <colgroup>
              <col className="w-full sm:w-1/2" />
            </colgroup>
            <tbody>
              {news.map((news, index) => (
                <tr
                  key={news.id}
                  className={`border-b ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
                >
                  <td
                    className={`max-w-0 border-none ${
                      index === 0 ? 'pt-0 pb-5' : 'py-5'
                    } pl-4 pr-3 text-sm sm:pl-0`}
                  >
                    <div
                      style={{ fontSize: '18px' }}
                      className={`${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
                    >
                      {news.name}
                    </div>
                    <div className="mt-1 truncate">
                      <a
                        className="text-blue-500 hover:text-blue-700"
                        target="blank"
                        href={news.url}
                      >
                        {news.meta}
                      </a>
                    </div>
                    <div className={`mt-1 truncate text-gray-500`}>{news.description}</div>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}
