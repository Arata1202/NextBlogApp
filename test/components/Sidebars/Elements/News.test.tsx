import { render } from '@testing-library/react';
import News from '@/components/Sidebars/Elements/News';

describe('News', () => {
  test('スナップショット（News）', () => {
    const { asFragment } = render(<News />);
    expect(asFragment()).toMatchSnapshot();
  });
});
