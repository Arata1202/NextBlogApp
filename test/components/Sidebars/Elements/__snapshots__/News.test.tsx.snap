// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News スナップショット（News） 1`] = `
<DocumentFragment>
  <div
    class="pt-8 px-4 border py-5 mt-5 LightTheme"
  >
    <h1
      class="profile text-2xl text-center font-semibold flex justify-center"
    >
      <svg
        aria-hidden="true"
        class="h-8 w-8 mr-2"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.125 3C3.089 3 2.25 3.84 2.25 4.875V18a3 3 0 0 0 3 3h15a3 3 0 0 1-3-3V4.875C17.25 3.839 16.41 3 15.375 3H4.125ZM12 9.75a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5H12Zm-.75-2.25a.75.75 0 0 1 .75-.75h1.5a.75.75 0 0 1 0 1.5H12a.75.75 0 0 1-.75-.75ZM6 12.75a.75.75 0 0 0 0 1.5h7.5a.75.75 0 0 0 0-1.5H6Zm-.75 3.75a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5H6a.75.75 0 0 1-.75-.75ZM6 6.75a.75.75 0 0 0-.75.75v3c0 .414.336.75.75.75h3a.75.75 0 0 0 .75-.75v-3A.75.75 0 0 0 9 6.75H6Z"
          fill-rule="evenodd"
        />
        <path
          d="M18.75 6.75h1.875c.621 0 1.125.504 1.125 1.125V18a1.5 1.5 0 0 1-3 0V6.75Z"
        />
      </svg>
      お知らせ
    </h1>
    <div
      class=""
    >
      <div
        class="mt-5"
        style="max-height: 240px; overflow-y: auto;"
      >
        <table
          class="min-w-full"
        >
          <colgroup>
            <col
              class="w-full sm:w-1/2"
            />
          </colgroup>
          <tbody>
            <tr
              class="border-b LightTheme"
            >
              <td
                class="max-w-0 border-none pt-0 pb-5 pl-4 pr-3 text-sm sm:pl-0"
              >
                <div
                  class="LightTheme"
                  style="font-size: 18px;"
                >
                  ダークモードに対応しました。画面右上から切り替えることができます。
                </div>
                <div
                  class="mt-1 truncate"
                >
                  <a
                    class="text-blue-500 hover:text-blue-700"
                    target="blank"
                  />
                </div>
                <div
                  class="mt-1 truncate text-gray-500"
                >
                  2024年11月7日
                </div>
              </td>
            </tr>
            <tr
              class="border-b LightTheme"
            >
              <td
                class="max-w-0 border-none py-5 pl-4 pr-3 text-sm sm:pl-0"
              >
                <div
                  class="LightTheme"
                  style="font-size: 18px;"
                >
                  Android版アプリ「リアル大学生：モバイル」がリリースされました。
                </div>
                <div
                  class="mt-1 truncate"
                >
                  <a
                    class="text-blue-500 hover:text-blue-700"
                    href="https://play.google.com/store/apps/details?id=com.realunivlog.flutterblogapp"
                    target="blank"
                  >
                    Google Playからダウンロードする
                  </a>
                </div>
                <div
                  class="mt-1 truncate text-gray-500"
                >
                  2024年9月30日
                </div>
              </td>
            </tr>
            <tr
              class="border-b LightTheme"
            >
              <td
                class="max-w-0 border-none py-5 pl-4 pr-3 text-sm sm:pl-0"
              >
                <div
                  class="LightTheme"
                  style="font-size: 18px;"
                >
                  RSSの不具合を修正しました。お手数ですが、新しいURLから再登録をお願いします。
                </div>
                <div
                  class="mt-1 truncate"
                >
                  <a
                    class="text-blue-500 hover:text-blue-700"
                    href="https://meta.realunivlog.com/rss.xml"
                    target="blank"
                  >
                    RSSを登録する
                  </a>
                </div>
                <div
                  class="mt-1 truncate text-gray-500"
                >
                  2024年9月30日
                </div>
              </td>
            </tr>
            <tr
              class="border-b LightTheme"
            >
              <td
                class="max-w-0 border-none py-5 pl-4 pr-3 text-sm sm:pl-0"
              >
                <div
                  class="LightTheme"
                  style="font-size: 18px;"
                >
                  iOS版アプリ「リアル大学生：モバイル」がリリースされました。
                </div>
                <div
                  class="mt-1 truncate"
                >
                  <a
                    class="text-blue-500 hover:text-blue-700"
                    href="https://apps.apple.com/jp/app/リアル大学生-モバイル/id6590619103"
                    target="blank"
                  >
                    App Storeからダウンロードする
                  </a>
                </div>
                <div
                  class="mt-1 truncate text-gray-500"
                >
                  2024年8月3日
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
